forch/device_report_server.py

```diff
@@ -44,12 +74,20 @@ def __init__(self, on_receiving_result, address=None, port=None, max_workers=Non
         self._server = grpc.server(
             futures.ThreadPoolExecutor(max_workers=max_workers or MAX_WORKERS_DEFAULT))
 
-        servicer = DeviceReportServicer(on_receiving_result)
-        device_report_pb2_grpc.add_DeviceReportServicer_to_server(servicer, self._server)
+        self._servicer = DeviceReportServicer(on_receiving_result, )
```

Comment:
I'm curious what the comma does... isn't that a syntax error?

Reply:
I removed the comma but that's not a syntax error in python; the same way a tuple can have an extra comma in the end e.g. (1,2,)